packages/core/src/utils-hoist/requestdata.ts

```diff
@@ -295,8 +295,8 @@ export function addNormalizedRequestDataToEvent(
 
     if (Object.keys(extractedUser).length) {
       event.user = {
-        ...event.user,
         ...extractedUser,
+        ...event.user,
```

Comment:
This is pretty weird and IMHO wrong before - if a user is set explicitly (e.g. via `Sentry.setUser()` this should always take precedence over this integration handling, I'd say.